src/directives/float-label/float-label.directive.ts

```diff
@@ -0,0 +1,71 @@
+import { Directive, Input, OnInit, TemplateRef, Renderer2, OnDestroy, HostBinding, OnChanges, ElementRef } from '@angular/core';
+
+@Directive({
+    selector: '[uxFloatLabel]',
+    host: {
+        'class': 'ux-float-label'
+    }
+})
+export class FloatLabelDirective implements OnInit, OnChanges, OnDestroy {
+
+    @Input('uxFloatLabel')
+    input: HTMLInputElement;
+
+    @HostBinding('class.ux-float-label-raised')
+    @Input()
+    raised: boolean = false;
+
+    @Input()
+    mode: 'focus' | 'input' = 'focus';
+
+    private _eventHandles: any[] = [];
+
+    constructor(private _elementRef: ElementRef, private _renderer: Renderer2) { }
+
+    ngOnInit(): void {
+        this._eventHandles.push(
+            this._renderer.listen(this.input, 'focus', this.inputFocus.bind(this))
+        );
+        this._eventHandles.push(
+            this._renderer.listen(this.input, 'blur', this.inputBlur.bind(this))
+        );
+        this._eventHandles.push(
+            this._renderer.listen(this.input, 'input', this.inputChange.bind(this))
+        );
+
+        // Check initial input value
+        this.raised = !!this.input.value;
```

Comment:
In plunker I tried to initially set `[raised]="true"`, this seemed to have no effect, I assume this might be because raised is getting overridden by this line here?

Reply:
Ah right, I see what you mean now. That's fine then, I can't really think of any better way. Just wanted to run it by you and make sure this was expected